### PR TITLE
Remove target-context flags

### PR DESCRIPTION
**Pull request by ulyssessouza**
As a user of the Docker App CLI tool
So that I have a consistent user experience with the Docker CLI
I want the target context flag removed

**- What I did**
Remove all the `--target-context` and references to it in the README.md

**- How I did it**
- removed all the `--target-context` and references to it in the README.md
- app is installed in context pointed by docker CLI
- installer image runs in `default` context by defaut until overriden by `--installer-context`

**- How to verify it**
Check the commands to see that the flags are not there anymore

**- Description for the changelog**
Remove target-context flags

**- A picture of a cute animal (not mandatory but encouraged)**
2baby-goats

### CODE REVIEW

File: internal/commands/cnab.go [hunk 1]
```diff
@@ -157,7 +157,7 @@ func prepareCredentialSet(b *bundle.Bundle, opts ...credentialSetOpt) (map[strin
 	_, requiresDockerContext := b.Credentials[internal.CredentialDockerContextName]
 	_, hasDockerContext := creds[internal.CredentialDockerContextName]
 	if requiresDockerContext && !hasDockerContext {
-		return nil, errors.New("no target context specified. Use --target-context= or DOCKER_TARGET_CONTEXT= to define it")
+		return nil, errors.New("no target context specified. Use DOCKER_TARGET_CONTEXT= to define it")
```
Comment (silvin-lubecki) [inline internal/commands/cnab.go]: The error still mentions "target context specified"

Comment (silvin-lubecki) [reply internal/commands/cnab.go]: Do we really still want to advertise about this `DOCKER_TARGET_CONTEXT` feature?

File: internal/commands/list.go [hunk 2]
```diff
@@ -51,14 +55,13 @@ func listCmd(dockerCli command.Cli) *cobra.Command {
...
 	}
-	cmd.Flags().StringVar(&opts.targetContext, "target-context", "", "List installations on this context")
 
 	return cmd
 }
 
 func runList(dockerCli command.Cli, opts listOptions) error {
-	targetContext := getTargetContext(opts.targetContext, dockerCli.CurrentContext())
-	installations, err := getInstallations(targetContext, config.Dir())
+	opts.SetDefaultTargetContext(dockerCli)
```
Comment (silvin-lubecki) [inline internal/commands/list.go]: Can't we simplify it with just calling ?
```go
targetContext := getTargetContext(dockerCli.CurrentContext())
```
And by the way simplify `getTargetContext` func too?

**Comment by kinghuang** [comment]
I strongly object to this change, assuming docker-app continues to spin up a container with the invocation image in order to do installs/uninstalls.

In my environment, it is not possible to run arbitrary containers on the swarm manager nodes. With docker-app 0.8.0, I've had to make sure that my Docker context is set to my local Docker engine (e.g., DOCKER_CONTEXT=default), and run docker-app commands with `--target-context=the-remote-cluster`. Otherwise, it attempts to run the invocation container on the manager node and fails.

Removing `--target-context` will completely break my ability to deploy apps, if an invocation container is needed.

**Comment by ulyssessouza** [comment]
@kinghuang Thank you for you feedback!
What do you think about instead of this, we use `--context` as the target and introducing a new experimental flag called `--invocation-context`?

So in your example, you could just use something like:
```
$ docker app install --context the-remote-cluster --invocation-context default myapp
```

**Comment by kinghuang** [comment]
Yes, I believe an `--invocation-context` option will work, and achieves the goal of consistency with the rest of the Docker CLI. An associated environment variable for the option would be greatly appreciated, too!

File: internal/commands/root.go [hunk 3]
```diff
@@ -121,21 +140,53 @@ func (o *parametersOptions) addFlags(flags *pflag.FlagSet) {
...
 }
 
 func getTargetContext(optstargetContext, currentContext string) string {
 	var targetContext string
 	switch {
 	case optstargetContext != "":
 		targetContext = optstargetContext
-	case os.Getenv("DOCKER_TARGET_CONTEXT") != "":
-		targetContext = os.Getenv("DOCKER_TARGET_CONTEXT")
+	case os.Getenv("INSTALLER_TARGET_CONTEXT") != "":
```
Comment (silvin-lubecki) [inline internal/commands/root.go]: nit: DOCKER_INSTALLER_CONTEXT?

File: README.md [hunk 4]
```diff
@@ -188,23 +188,6 @@ $ docker app inspect myrepo/hello:0.1.0
 **Note**: Commands like `install`, `upgrade`, `render`, etc. can also be used
 directly on Application Packages that are in a registry.
 
-You can specify the Docker endpoint where an application is installed using a
```
Comment (rumpl) [inline README.md]: We need some doc about the `--installer-context`

File: internal/commands/list.go [hunk 5]
```diff
@@ -17,7 +17,6 @@ import (
 )
 
 type listOptions struct {
```
Comment (rumpl) [inline internal/commands/list.go]: listOptions is not used any more right?

**Pull request merged**